src/Extension/SandboxExtension.php

```diff
@@ -50,16 +53,21 @@ public function disableSandbox()
         $this->sandboxed = false;
     }
 
-    public function isSandboxed()
+    public function isSandboxed($source = null)
     {
-        return $this->sandboxedGlobally || $this->sandboxed;
+        return $this->sandboxedGlobally || $this->sandboxed || $this->isSourceSandboxed($source);
     }
 
     public function isSandboxedGlobally()
     {
         return $this->sandboxedGlobally;
     }
 
+    public function isSourceSandboxed(?Source $source) : bool
```

Comment:
This should be a private method unless there is a good reason to expose that publicly (once something is public, we cannot remove it anymore)